torch/csrc/lazy/core/tensor_impl.cpp

```diff
@@ -83,6 +84,14 @@ LTCTensorImpl::LTCTensorImpl(LazyTensor&& tensor)
   // according to https://github.com/pytorch/xla/pull/2682.
   is_non_overlapping_and_dense_ = false;
   set_sizes_strides_policy(SizesStridesPolicy::CustomSizes);
+
+  auto rank = tensor_->shape().Get().sizes().size();
```

Comment:
this part looks pretty straight-forward to me.  anything in particular that you are concerned about?

Reply:
I also think it should be fine.